sqlanalytics/api/wrapper.go

```diff
@@ -0,0 +1,240 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/pkg/errors"
+)
+
+// Wrapper is an API wrapper for the SQL Analytics queries, visualizations, and dashboards API.
+type Wrapper struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+// NewWrapper creates and returns an API wrapper.
+func NewWrapper(ctx context.Context, m interface{}) Wrapper {
```

Comment:
Please split `Wrapper` into `VisualizationAPI`, `QueryAPI`, `WidgetAPI` structs and move entities and API wrappers to respective resource files, e.g. resource_widget.go will have `NewWidgetsAPI` factory, `WidgetsAPI` struct, `WidgetEntity` terraform entity for reflection, `Widget` api entity, and `ResourceWidget`. The main reason for this is that we have 82%+ code coverage requirement for all code that is written and `gotestsum` calculates coverage on the package level. I've figured it, when i was confident that [reflect_resource](https://app.codecov.io/gh/databrickslabs/terraform-provider-databricks/compare/553/tree/common/reflect_resource.go) is pretty well tested by other packages, but the coverage report didn't show any good results for it. [api package coverage](https://app.codecov.io/gh/databrickslabs/terraform-provider-databricks/compare/553/tree/sqlanalytics) is missing from the report, meaning that there are long-term risks of changes going untested to production. This may not be ideal, but it is consistent with the rest of the codebase. Even the special package, dedicated to [Accounts API only](https://github.com/databrickslabs/terraform-provider-databricks/tree/master/mws), follows the same conventions as packages that create resources within the workspace. _We may reconsider package layout conventions if we want to have databricks-go-sdk and refactor all 7 packages in a **separate** pull request, adjusting `resource.RetryContext` along the way._